### PR TITLE
Add requirement that secret inputs be negotiated.

### PR DESCRIPTION
**Pull request by jhoyla**

### CODE REVIEW

File: draft-jhoyla-tls-extended-key-schedule.md [hunk 1]
```diff
@@ -103,13 +104,21 @@ an appropriate frame, to the `0` input.
...
-# Key Schedule Extension Structure
+# Key Schedule Injection Negotiation
 
-In some cases, protocols may require more than one secret to be injected at a particular
-stage in the key schedule. Thus, we require a generic and extensible way of doing so.
-To accomplish this, we use a structure -- KeyScheduleInput -- that encodes well-ordered
-sequences of secret material to inject into the key schedule. KeyScheduleInput is defined
-as follows:
+Applications which make use of additional key schedule inputs MUST define an
+extension that negotiates the content and type of that input.  This input MUST
```
Comment (chris-wood) [inline draft-jhoyla-tls-extended-key-schedule.md]:
```suggestion
Applications which make use of additional key schedule inputs MUST define a
mechanism for negotiating the content and type of that input.  This input MUST
```

File: draft-jhoyla-tls-extended-key-schedule.md [hunk 2]
```diff
@@ -103,13 +104,21 @@ an appropriate frame, to the `0` input.
...
+be framed in a KeyScheduleSecret struct, as defined in {{structure}}.
+Applications must take care that the negotiation of the content of the secret
+inputs unambiguously defines a single input.
```
Comment (chris-wood) [inline draft-jhoyla-tls-extended-key-schedule.md]: I didn't parse this -- does this mean that applications can only negotiate one secret input at a time?

**Review by chris-wood — CHANGES_REQUESTED** [review]

File: draft-jhoyla-tls-extended-key-schedule.md (hunk 2 above)
Comment (jhoyla) [reply draft-jhoyla-tls-extended-key-schedule.md]: No, this was terribly phrased. I meant that any given negotiation must have an unambiguous conclusion. It must be impossible even under adversarial conditions for the client and server to see the same extensions but believe they have negotiated different keys.

Comment (chris-wood) [reply draft-jhoyla-tls-extended-key-schedule.md]: Perhaps we can rephrase it as such?

**Pull request merged**